Sources/Basic/FileSystem.swift

```diff
@@ -276,6 +239,57 @@ private class LocalFileSystem: FileSystem {
             break
         }
     }
+
+    /// An iterator to move though contents of a directory.
+    private final class DirectoryContentsIterator: Sequence, IteratorProtocol {
```

Comment:
I think this might be more complicated than it needs to be, can we not use `AnyIterator`?